Sources/AppCoinsSDK/Models/Services/Gamification/AppCoinGamificationServiceClient.swift

```diff
@@ -27,14 +27,21 @@ internal class AppCoinGamificationServiceClient : AppCoinGamificationService {
             let task = URLSession.shared.dataTask(with: request) { data, response, error in
                 if let error = error {
                     if let nsError = error as NSError?, nsError.code == NSURLErrorNotConnectedToInternet {
-                        result(.failure(.noInternet))
+                        result(.failure(.noInternet(message: "Internet Connection Failed", description: "Could not get internet connection to \(url)")))
```

Comment:
Also add code location on the description or service layer errors.

Reply:
Done